mongo/options/clientoptions.go

```diff
@@ -300,187 +322,305 @@ func (c *ClientOptions) ApplyURI(uri string) *ClientOptions {
+// should be validated. If true (e.g. "tlsInsecure=true"), the TLS library will accept any certificate presented by the
+// server and any host name in that certificate. Note that setting this to true makes TLS susceptible to
+// man-in-the-middle attacks and should only be done for testing.
+//
+// The default is nil, meaning no TLS will be enabled.
 func (c *ClientOptions) SetTLSConfig(cfg *tls.Config) *ClientOptions {
 	c.TLSConfig = cfg
 	return c
 }
 
-// SetWriteConcern sets the write concern.
+// SetWriteConcern specifies the write concern to use to configure the Client. This can also be se through the following
+// URI options:
+//
+// 1. "w": Specify the number of nodes in the cluster that must acknowledge write operations before the operation
+// returns or "majority" to specify that a majority of the nodes must acknowledge writes. This can either be an integer
+// (e.g. "w=10") or the string "majority" (e.g. "w=majority").
+//
+// 2. "wTimeoutMS": Specify how long write operations should wait for the correct number of nodes to acknowledge the
+// operation (e.g. "wTimeoutMS=1000").
+//
+// 3. "journal": Specifies whether or not write operations should be written to an on-disk journal on the server before
+// returning (e.g. "journal=true").
+//
+// The default is nil, meaning the server will use its configured default.
 func (c *ClientOptions) SetWriteConcern(wc *writeconcern.WriteConcern) *ClientOptions {
 	c.WriteConcern = wc
 
 	return c
 }
 
-// SetZlibLevel sets the level for the zlib compressor.
+// SetZlibLevel specifies the level for the zlib compressor. This option is ignored if zlib is not specified as a
+// compressor through ApplyURI or SetCompressors. Supported values are -1 through 9, inclusive. -1 tells the zlib
+// library to use its default, 0 means no compression, 1 means best speed, and 9 means best compression.
+// This can also be set through the "zlibCompressionLevel" URI option (e.g. "zlibCompressionLevel=-1"). Defaults to -1.
 func (c *ClientOptions) SetZlibLevel(level int) *ClientOptions {
 	c.ZlibLevel = &level
 
 	return c
 }
 
-// SetZstdLevel sets the level for the zstd compressor.
+// SetZstdLevel sets the level for the zstd compressor. This option is ignored if zstd is not specified as a compressor
+// through ApplyURI or SetCompressors. Supported values are 1 through 20, inclusive. 1 means best speed and 20 means
+// best compression. This can also be set through the "zstdCompressionLevel" URI option. Defaults to 6.
 func (c *ClientOptions) SetZstdLevel(level int) *ClientOptions {
 	c.ZstdLevel = &level
 	return c
 }
 
-// SetAutoEncryptionOptions specifies options used to configure automatic encryption.
+// SetAutoEncryptionOptions specifies an AutoEncryptionOptions instance to configure the Client to automatically
+// encrypt and decrypt collection commands and their results. See the options.AutoEncryptionOptions documentation
```

Comment:
Is it just collection commands?  Is it necessary to limit it here?

Reply:
This was taken from PyMongo. On second thought, I don't think it's necessary to limit it here as it's already documented that automatic encryption doesn't work for operations on databases or views in `AutoEncryptionOptions`. Removed.